roles/publish-docs/tasks/github.com/main.yml

```diff
@@ -0,0 +1,27 @@
+---
+# tasks file for publish-github
+- name: Create temporary build directory
+  tempfile:
+    state: directory
+    suffix: github-wiki
+  register: tempdir
+
+- name: Clone wiki repo
+  git:
+    repo: git@github.com:{{ github.repo_owner }}/{{ github.repo }}.wiki.git
+    dest: "{{ tempdir.path }}/{{ github.repo }}.wiki"
+    version: HEAD
+
+- name: Add conents to wiki file
+  copy:
+    content: "{{ generate_docs.output }}"
+    dest: "{{ tempdir.path }}/{{ github.repo }}.wiki/{{ github.edit_doc }}"
+
+# Seems like this has to use shell module due to use of chdir
+- name: Publish changes to wiki
+  shell: |
+    chdir={{ tempdir.path }}/{{ github.repo }}.wiki
+    git add {{ github.edit_doc }}
+    git commit -m "{{ github.commit_message }}"
+    git push
+  notify: github cleanup temp
```

Comment:
As discussed, this clean-up only happens if the task reports as changed. There's really no good solution for this, other than ignore error and always do the clean-up.... or alternatively don't use a file (if possible). ... or you can wrap everything in a `block` and use `rescue` to do the clean-up in case of an error, but then you need to do this twice .... so, not sure what solution to recommend here. Remind me to discuss this next time we chat.